src/common/chess-engine.ts

```diff
@@ -0,0 +1,53 @@
+// engine.ts
+
+import { Chess, Square } from "chess.js";
+import { GameFinishedReason } from "./game-end";
+
+export class ChessEngine {
+    private chess: Chess;
+
+    constructor() {
```

Comment:
This should take a `fen` position as an argument. If one isn't specified, it should default to `DEFAULT_POSITION`.